eclipse-theia/theia
# Allow gzip encoding for request services

#### What it does

Closes https://github.com/eclipse-theia/theia/issues/11233

Allow `gzip` encoding by default for all requests coming from the `NodeRequestService`. Sends the respective `Accept-Encoding` headers and respects the `content-encoding` response header.

#### How to test

1. Confirm (debug) that the headers are present on request and response headers.
2. `download:plugins` cli command should still work as expected (maybe a bit faster)

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
